cluster/gce/util.sh

```diff
@@ -364,14 +364,17 @@ function create-firewall-rule {
 # $4: The kube-env metadata.
 function create-node-template {
   detect-project
+  local template_name="$1"
+  # trim template name to pass gce name validation
+  template_name="$(echo $template_name | cut -c 1-63 | sed 's/-*$//g')"
```

Comment:
AFAIK, this is going to break because it's tightly coupled to `prepare-node-upgrade` and `do-node-upgrade` in `cluster/gce/upgrade.sh`.